src/main/java/com/blackducksoftware/integration/jira/task/issue/tracker/IssueTrackerTask.java

```diff
@@ -76,19 +76,20 @@ public Boolean call() throws Exception {
             final PluginConfigurationDetails configDetails = new PluginConfigurationDetails(settings);
             final JiraSettingsService jiraSettingsService = new JiraSettingsService(settings);
 
+            final Optional<BlackDuckJiraConfigSerializable> config = createJiraConfig(configDetails);
+            if (config.filter(jiraConfig -> jiraConfig.getHubProjectMappings() == null || jiraConfig.getHubProjectMappings().isEmpty()).isPresent()) {
```

Comment:
Wouldn't the optional be empty if there are no mappings, based on the way you deserialize the config?

Reply:
Correct we had the deserialization code in two places.  So I combined it into this class.  The two places did the deserialization slightly differently and upon investigation it didn't make sense.  It should be consistent to deserialize the data. 
The BlackDuckJiraTask would deserialize the config.  If there are no project mappings then we shouldn't do anything.  It used to rely on null being returned by the method.  So I switched it to Optional since that is better than null checks.

IssueTracker would deserialize only part of the configuration data.  It was inconsistent with BlackDuckJiraTask.  It would not deserialize the project mappings.  If the project mapping has been removed for the project we don't want to track it anymore.  So now we deserialize the data in a consistent way.  

That kept the existing functionality intact.  However looking at the code and explaining this here.  It might make more sense to add a method to BlackDuckJiraConfigSerializable to see if there are project mappings and use that to key off.  Always return a BlackDuckJiraConfigSerializable and then check if the mappings are empty.

Reply:
The reason I asked is, it seems this if check is to check if the mapping is empty. If that is the case, then we should just check config.isEmpty() and that should accomplish the same thing since we know the optional will be empty if there are no mappings